File: frontend/src/api/emergency.js
import { apiClient } from "./client";

/**
 * GET /emergency/nearby-hospitals - confirmed against
 * app/models/emergency.py / app/routes/emergency.py. No auth required.
 *
 * Returns: [{ name, latitude, longitude, distance_km, address, phone }]
 */
export async function getNearbyHospitals(latitude, longitude, { radiusKm = 5 } = {}) {
  const { data } = await apiClient.get("/emergency/nearby-hospitals", {
    params: { lat: latitude, lon: longitude, radius_km: radiusKm },
  });
  return data;
}

File: app/models/emergency.py
"""
Data models for the emergency-flow "nearby hospitals" feature
(GET /emergency/nearby-hospitals).
"""

from pydantic import BaseModel, Field


class NearbyHospital(BaseModel):
    """One hospital/clinic returned by the nearby-hospitals lookup."""

    name: str
    latitude: float
    longitude: float
    distance_km: float = Field(..., description="Straight-line distance from the queried point.")
    address: str | None = Field(default=None, description="Best-effort address from source tags.")
    phone: str | None = None

    model_config = {
        "json_schema_extra": {
            "example": {
                "name": "City General Hospital",
                "latitude": 23.2599,
                "longitude": 77.4126,
                "distance_km": 1.4,
                "address": "MG Road, Bhopal",
                "phone": None,
            }
        }
    }
